src/common/config/manager.go

```diff
@@ -0,0 +1,218 @@
+}
+
+// RegistDriver ...
+func RegistDriver(driverType string, d Driver) {
+	driverRegistry[driverType] = d
+}
+
+// GetStore ...
+func GetStore(storeType string) Store {
+	return storeRegistry[storeType]
+}
+
+// CfgManagerInstance ... Single instance of configure manager,
+// there is only one copy of CfgManagerInstance in an env
+func CfgManagerInstance() *CfgManager {
+	initLocalManagerOnce.Do(func() {
+		_managerInstance = newCfgManager()
+		_managerInstance.init(LocalStore)
+	})
+	return _managerInstance
+}
+
+// CfgManagerRemoteInstance - Single instance of configure manager
+// There is only one copy of CfgManagerInstance in an env
+func CfgManagerRemoteInstance() *CfgManager {
+	initRemoteManagerOnce.Do(func() {
+		_managerInstance = newCfgManager()
+		_managerInstance.init(RemoteStore)
+	})
+	return _managerInstance
+}
+
+func newCfgManager() *CfgManager {
+	return &CfgManager{}
+}
+
+func (c *CfgManager) init(storeName string) {
+	c.store = GetStore(storeName)
+	c.driver = GetDriver()
+	// load default value
+	c.LoadDefaultValues()
+	// load system config from env
+	c.LoadSystemConfigFromEnv()
+}
+func (c *CfgManager) readMap(key string) (*ConfigureValue, error) {
+	return c.store.Get(key)
+}
+
+func (c *CfgManager) writeMap(key string, value ConfigureValue) {
+	c.store.Set(key, value)
+}
+
+// LoadDefaultValues ...
+func (c *CfgManager) LoadDefaultValues() {
+	// Init Default Value
+	itemArray := MetaDataInstance().GetAllMetaData()
+	for _, item := range itemArray {
+		// Every string type have default value, other types should have a default value
+		if _, ok := item.ItemType.(*StringType); ok || len(item.DefaultValue) > 0 {
+			cfgValue := ConfigureValue{item.Name, item.DefaultValue}
+			err := cfgValue.Validate()
+			if err == nil {
+				c.writeMap(item.Name, cfgValue)
+			} else {
+				log.Errorf("Failed to init config item %+v, default err: %+v", c, err)
+			}
+		}
+	}
+}
+
+// LoadSystemConfigFromEnv ...
+func (c *CfgManager) LoadSystemConfigFromEnv() {
+	itemArray := MetaDataInstance().GetAllMetaData()
+	// Init System Value
+	for _, item := range itemArray {
+		if item.Scope == SystemScope && len(item.EnvKey) > 0 {
+			if envValue, ok := os.LookupEnv(item.EnvKey); ok {
+				configValue := ConfigureValue{item.Name, envValue}
+				err := configValue.Validate()
+				if err == nil {
+					c.store.Set(item.Name, configValue)
+					return
+				}
+				log.Errorf("Failed to init system config item %+v,  err: %+v", c, err)
+			}
+		}
+	}
+}
+
+// GetAll ... Get all settings
+func (c *CfgManager) GetAll() []ConfigureValue {
+	c.driver.Load(c.store)
+	results := make([]ConfigureValue, 0)
+	metaDataList := MetaDataInstance().GetAllMetaData()
+	for _, item := range metaDataList {
+		if cfgValue, err := c.store.Get(item.Name); err == nil {
+			results = append(results, *cfgValue)
+		}
+	}
+	return results
+}
+
+// Load - Load configuration from storage, like database or redis
+func (c *CfgManager) Load() error {
```

Comment:
load/save/set/get these functions are confusing by name, again, I thins no need to hole the reference of driver in the manager.